soc/arm/nxp_lpc/lpc55xxx/Kconfig.soc

```diff
@@ -7,6 +7,16 @@ choice
 	prompt "LPC5500 Series MCU Selection"
 	depends on SOC_SERIES_LPC55XXX
 
+config SOC_LPC55S16
+	bool "SOC_LPC55S16 M33"
+	select CPU_CORTEX_M33
+	select CPU_HAS_ARM_SAU
+	select CPU_HAS_ARM_MPU
+	select CPU_HAS_FPU
+	select ARMV8_M_DSP
+	select ARM_TRUSTZONE_M
+	select HAS_MCUX_IAP if TRUSTED_EXECUTION_SECURE
```

Comment:
This is a bit dodgy, I know. The NXP HAL defines `BOOTLOADER_API_TREE_POINTER` to `((bootloader_tree_t *)0x130010f0U)` thus the IAP flash driver fails horribly if operating in non-secure mode.

Reply:
Hmmm, not sure what to suggest on this one...

Reply:
what this be better as if !NON_SECURE?

Reply:
I have updated the LPC55S16 `HAS_MCUX_IAP` to only be selected if `!TRUSTED_EXECUTION_NONSECURE`.

I think we could even change this to have `HAS_MCUX_IAP` depend on `!TRUSTED_EXECUTION_NONSECURE`, since this is a pre-requisite of the current MCUXpresso IAP flash driver. Opinions?

Reply:
@MaureenHelm I have re-requested your review on this due to the above change and suggested change.